src/main/java/com/matchbook/sdk/clients/rest/UserResponseImpl.java

```diff
@@ -0,0 +1,74 @@
+package com.matchbook.sdk.clients.rest;
+
+import java.io.IOException;
+
+import com.fasterxml.jackson.databind.ObjectReader;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import com.matchbook.sdk.StreamObserver;
+import com.matchbook.sdk.clients.rest.dtos.user.LoginRequest;
+import com.matchbook.sdk.clients.rest.dtos.user.LoginResponse;
+import com.matchbook.sdk.clients.rest.dtos.user.LogoutResponse;
+import com.matchbook.sdk.configs.ClientConnectionManager;
+import com.matchbook.sdk.exceptions.MatchbookSDKHTTPException;
+import com.squareup.okhttp.Callback;
+import com.squareup.okhttp.MediaType;
+import com.squareup.okhttp.Request;
+import com.squareup.okhttp.RequestBody;
+import com.squareup.okhttp.Response;
+import com.squareup.okhttp.ResponseBody;
+
+public class UserResponseImpl implements UserResource {
```

Comment:
Maybe you intended `UserResourceImpl` instead of `UserResponseImpl`?

BTW, I've never liked too much the "convention" `InterfaceName` vs `InterfaceNameImpl` for concrete implementations... do you know if there is a better guideline for these names?

Reply:
I know it was me suggesting the name `Resource`, though I have finally realised what your intentions were and don't like the name any more:sweat_smile: 
What about `RestClient`? For example, in this case, `UserRestClient`?

Reply:
Yeah, usually i'm using X and XImpl, if you have any suggestions we can use it

Reply:
We'll think more about it. In the meantime, here's a discussion on this on [SO](https://stackoverflow.com/questions/541912/interface-naming-in-java).